engine/src/flutter/lib/web_ui/lib/src/engine/html/dom_canvas.dart

```diff
@@ -351,8 +350,8 @@ SVGSVGElement pathToSvgElement(SurfacePath path, SurfacePaintData paint) {
           paint.strokeWidth != null)) {
     svgPath.setAttribute('stroke', colorValueToCssString(paint.color));
     svgPath.setAttribute('stroke-width', '${paint.strokeWidth ?? 1.0}');
-    if (paint.strokeCap != null) {
-      svgPath.setAttribute('stroke-linecap', '${stringForStrokeCap(paint.strokeCap)}');
+    if (paint.strokeCap case final value?) {
+      svgPath.setAttribute('stroke-linecap', value);
```

Comment:
I'm not too familiar with this new syntax but isn't `paint.strokeCap` a `ui.StrokeCap`? We should pass in a `String` here

Reply:
yep! great catch!